### PR TITLE
App Config Spring - Snapshot references

### PR DESCRIPTION
**Pull request by mrm9084**
# Description

Adds support for Snapshot References

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [X] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationApplicationSettingPropertySource.java [hunk 1]
```diff
@@ -110,6 +123,30 @@ protected void processConfigurationSettings(List<ConfigurationSetting> settings,
...
+        WatchedConfigurationSettings featureFlags = new WatchedConfigurationSettings(null, featureFlagsList);
+        featureFlagClient.processFeatureFlags(featureFlags, replicaClient.getEndpoint());
+    }
+
+    private List<ConfigurationSetting> resolveSnapshotReferences(List<ConfigurationSetting> settings) {
+        List<ConfigurationSetting> resolvedSettings = new ArrayList<>();
+        for (ConfigurationSetting setting : settings) {
+            if (SNAPSHOT_REF_CONTENT_TYPE.equals(setting.getContentType())) {
+                // Handle snapshot reference
+                replicaClient.getTracingInfo().setUsesSnapshotReference();
```
[the remark this answers is not in the document]
Comment (mrm9084) [reply sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationApplicationSettingPropertySource.java:139]: We want to match the service as this is consistant.

File: sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationApplicationSettingPropertySource.java [hunk 2]
```diff
@@ -110,6 +126,30 @@ protected void processConfigurationSettings(List<ConfigurationSetting> settings,
...
+        featureFlagClient.processFeatureFlags(featureFlags, replicaClient.getEndpoint());
+    }
+
+    private List<ConfigurationSetting> resolveSnapshotReferences(List<ConfigurationSetting> settings, Context context) {
+        List<ConfigurationSetting> resolvedSettings = new ArrayList<>();
+        for (ConfigurationSetting setting : settings) {
+            if (SNAPSHOT_REF_CONTENT_TYPE.equals(setting.getContentType())) {
+                // Handle snapshot reference
+                replicaClient.getTracingInfo().setUsesSnapshotReference();
+                List<ConfigurationSetting> snapshotSettings = replicaClient.listSettingSnapshot(setting.getValue(),
```
Comment (moarychan) [inline sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationApplicationSettingPropertySource.java:140]: [Microsoft Learn](https://learn.microsoft.com/en-us/azure/azure-app-configuration/concept-snapshot-references#example-snapshot-reference) defines the reference value as `{"snapshot_name":"snapshot1"}`. `ConfigurationClient` expects only the snapshot name and sends it directly as the `snapshot` query parameter. Passing `setting.getValue()` therefore sends the entire JSON as the name. Please parse and validate `snapshot_name`, and update the tests to use the documented format.

File: sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationApplicationSettingPropertySource.java [hunk 3]
```diff
@@ -110,6 +126,30 @@ protected void processConfigurationSettings(List<ConfigurationSetting> settings,
...
+                    context);
```
Comment (moarychan) [inline sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationApplicationSettingPropertySource.java:141]: The `context` is passed into `listSettingSnapshot` and is correctly used by `getSnapshotWithResponse`. However, the following settings request uses `listConfigurationSettingsForSnapshot(snapshotName)`, which falls back to `Context.NONE`. Please use the three-argument overload so the first and subsequent settings pages keep the same startup/refresh tracing context, and verify this with `same(context)`.

File: sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationApplicationSettingPropertySource.java [hunk 4]
```diff
@@ -110,6 +126,30 @@ protected void processConfigurationSettings(List<ConfigurationSetting> settings,
...
+        WatchedConfigurationSettings featureFlags = new WatchedConfigurationSettings(null, featureFlagsList);
+        featureFlagClient.processFeatureFlags(featureFlags, replicaClient.getEndpoint());
+    }
+
+    private List<ConfigurationSetting> resolveSnapshotReferences(List<ConfigurationSetting> settings, Context context) {
+        List<ConfigurationSetting> resolvedSettings = new ArrayList<>();
+        for (ConfigurationSetting setting : settings) {
+            if (SNAPSHOT_REF_CONTENT_TYPE.equals(setting.getContentType())) {
+                // Handle snapshot reference
+                replicaClient.getTracingInfo().setUsesSnapshotReference();
```
Comment (moarychan) [inline sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationApplicationSettingPropertySource.java:139]: `usesSnapshotReference` is set here but never reset. Later non-snapshot requests from the same replica client can therefore still report `Features=SnapshotRef`. Please scope the flag to the snapshot requests, clear it in a `finally` block, and test both successful and failed loads.

File: sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationApplicationSettingPropertySource.java [hunk 5]
```diff
@@ -110,6 +126,30 @@ protected void processConfigurationSettings(List<ConfigurationSetting> settings,
                 properties.put(key, setting.getValue());
             }
         }
+
+        WatchedConfigurationSettings featureFlags = new WatchedConfigurationSettings(null, featureFlagsList);
+        featureFlagClient.processFeatureFlags(featureFlags, replicaClient.getEndpoint());
```
Comment (moarychan) [inline sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationApplicationSettingPropertySource.java:131]: `FeatureFlagClientTest#loadFeatureFlagsTestMultipleLoads` shows that loading `Alpha/Beta` and then `Alpha/Gamma` leaves all three flags in the map. That merge behavior is useful within one load, but a reference switch should unload flags owned only by the old snapshot. Please add a two-refresh regression test and replace flags at the correct source or full-load boundary without clearing other labels, selectors, or stores.

File: sdk/spring/spring-cloud-azure-appconfiguration-config/CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,8 @@
 
 ### Features Added
 
+- Added support for Snapshot References in App Configuration stores. Snapshot reference settings are now automatically resolved, loading the referenced snapshot's configuration settings as properties.
```
Comment (moarychan) [inline sdk/spring/spring-cloud-azure-appconfiguration-config/CHANGELOG.md:7]: please sync with main branch and append the changelog to new section `7.5.0-beta.1 (Unreleased)`.

Comment (moarychan) [inline sdk/spring/spring-cloud-azure-appconfiguration-config/CHANGELOG.md:7]: There's also the maintenance of Spring Cloud Azure version 6.x, please note that port this feature to [6.x branch](https://github.com/Azure/azure-sdk-for-java/tree/feature/spring-cloud-azure-6.x).

**Review by moarychan — CHANGES_REQUESTED** [review]

**Comment by moarychan** [comment]
I would like to add a new feature [appconfiguration/spring-cloud-azure-appconfiguration-config](https://github.com/Azure-Samples/azure-spring-boot-samples/blob/main/appconfiguration/spring-cloud-azure-starter-appconfiguration-config/spring-cloud-azure-starter-appconfiguration-config-sample/src/main/resources/application.propertiesl) to use the new feature in this PR.

File: sdk/spring/spring-cloud-azure-appconfiguration-config/CHANGELOG.md (hunk 6 above)
Comment (mrm9084) [reply sdk/spring/spring-cloud-azure-appconfiguration-config/CHANGELOG.md:7]: App Configuration historically only supports new features on the latest version unless a customer requests them to be back ported.

**Comment by mrm9084** [comment]
> I would like to add a new feature [appconfiguration/spring-cloud-azure-appconfiguration-config](https://github.com/Azure-Samples/azure-spring-boot-samples/blob/main/appconfiguration/spring-cloud-azure-starter-appconfiguration-config/spring-cloud-azure-starter-appconfiguration-config-sample/src/main/resources/application.propertiesl) to use the new feature in this PR.

I'm unsure what you mean by this.
